src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,135 @@
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.exception.DukeException;
+import duke.exception.DukeExceptionType;
+
+/**
+ * Handles all parsing of input by user into commands, description and date (if applicable)
+ */
+public class Parser {
+
+    /**
+     * Determines command given by user input
+     *
+     * @param text Input given by user
+     * @return Command to be supplied to be executed
+     * @throws DukeException If given an empty description, empty selection, invalid integer, unrecognized command
+     */
+    public static Command parse(String text) throws DukeException {
+        String command = command(text);
+        String description = description(text);
+        Command commandType;
+
+        // Throws exception for ToDo, find, Event and Deadline commands
+        if (command.equals("todo") || command.equals("event") || command.equals("deadline")) {
+            if (description.equals(command)) {
+                //Empty description
+                throw new DukeException(command, DukeExceptionType.EMPTY_DESCRIPTION);
+            }
+        }
+
+        // Throws exception for find command
+        if (command.equals("find")) {
+            if (description.equals(command)) {
+                //Empty keyword
+                throw new DukeException(command, DukeExceptionType.EMPTY_KEYWORD);
+            }
+        }
+
+        // Throws exception for done and delete commands
+        if (command.equals("done") || command.equals("delete")) {
+            if (description.equals(command)) {
+                throw new DukeException(command, DukeExceptionType.EMPTY_SELECTION);
+            } else if (!Utility.isNumeric(description)) {
+                // Selection not numeric
+                throw new DukeException(command, DukeExceptionType.INVALID_INTEGER);
+            }
+        }
+
+        switch (command) {
+        case "find":
+            commandType = new FindCommand(command, description);
+            break;
+        case "list":
+            commandType = new ListCommand(command);
+            break;
+        case "done":
+            commandType = new DoneCommand(command, description);
+            break;
+        case "delete":
+            commandType = new DeleteCommand(command, description);
+            break;
+        case "bye":
+            commandType = new ByeCommand(command);
+            break;
+        case "todo":
+            commandType = new AddCommand(command, description);
+            break;
+        case "event":
+        case "deadline":
+            String date = date(text);
+            commandType = new AddCommand(command, description, date);
+            break;
+        default:
+            throw new DukeException(DukeExceptionType.UNKNOWN_INPUT);
+        }
+        return commandType;
+    }
+
+    private static String command(String text) {
```

Comment:
Would getCommand or extractCommand be a better naming? 
I've also noticed the same issue in other methods. For example, the 'description', 'date' methods. It might be a bit confusing without a verb in the method name :)

Reply:
I do agree with yutingzou on this

Reply:
Hello, thank you for reviewing my code! Yes I agree, I'll work on it! :)